### PR TITLE
Moodle 4.0 refactoring

### PR DESCRIPTION
**Pull request by Syxton**

### CODE REVIEW

File: version.php [hunk 1]
```diff
@@ -25,7 +25,7 @@
 defined('MOODLE_INTERNAL') || die;
 
 $plugin->version = 2022031500;
-$plugin->requires  = 2020061500;
+$plugin->requires  = 2021052500;
```
Comment (PhMemmel) [inline version.php:28]: Do we have to change this for avoiding that this branch is being installed on a moodle <4.0?

Comment (Syxton) [reply version.php:28]: @PhMemmel  Yes, we need to change the requires for the 4.0 branch.  I am going to rename master to stable_39_311 and then rename the 4.0 branch to master.  Then I can close out this pull request.

Comment (PhMemmel) [reply version.php:28]: Can you just push this and do the branch-management or do you want me to do this? :-)

**Comment by PhMemmel** [comment]
@Syxton Just realized this branch still breaks tiles format, so I probably will have to add a patch for downwards compatibility. I can do this after reorganizing branches if you want

**Comment by PhMemmel** [comment]
Here we go, added a little bit of legacy code :-)

**Comment by PhMemmel** [comment]
@Syxton Thank you, awesome! Let's do some additional testing and after that, pull a release to get the early bird badge ;-)

**Pull request closed without merging**